specs/Print.md

```diff
@@ -0,0 +1,861 @@
+
+  CHECK_FAILURE(printSettings2->put_Orientation(printInfo.Layout));
+  CHECK_FAILURE(printSettings2->put_Copies(printInfo.Copies));
+  CHECK_FAILURE(printSettings2->put_PagesPerSheet(printInfo.PagesPerSheet));
+  CHECK_FAILURE(printSettings2->put_PageRanges(printInfo.Pages.c_str()));
+  if (printInfo.Media == COREWEBVIEW2_PRINT_MEDIA_SIZE_CUSTOM)
+  {
+    CHECK_FAILURE(printSettings2->put_PageWidth(printInfo.PaperWidth));
+    CHECK_FAILURE(printSettings2->put_PageHeight(printInfo.PaperHeight));
+  }
+  CHECK_FAILURE(printSettings2->put_ColorMode(printInfo.ColorMode));
+  CHECK_FAILURE(printSettings2->put_Collation(printInfo.Collation));
+  CHECK_FAILURE(printSettings2->put_Duplex(printInfo.Duplex));
+  CHECK_FAILURE(printSettings2->put_ScaleFactor(printInfo.ScaleFactor));
+  CHECK_FAILURE(printSettings2->put_ShouldPrintBackgrounds(printInfo.PrintBackgrounds));
+  CHECK_FAILURE(printSettings2->put_ShouldPrintHeaderAndFooter(printInfo.HeaderAndFooter));
+  CHECK_FAILURE(printSettings2->put_HeaderTitle(printInfo.HeaderTitle.c_str()));
+  CHECK_FAILURE(printSettings2->put_FooterUri(printInfo.FooterUri.c_str()));
+  CHECK_FAILURE(printSettings2->put_PrinterName(printerName.c_str()));
+
+  wil::unique_cotaskmem_string title;
+  CHECK_FAILURE(m_webView->get_DocumentTitle(&title));
+
+  CHECK_FAILURE(webView2_15->Print(
+      printSettings2.get(),
+      Callback<ICoreWebView2PrintCompletedHandler>(
+          [title = title.get(),
+          appWindow = m_appWindow](HRESULT errorCode, BOOL isSuccessful) -> HRESULT
+          {
+            std::wstring message = L"";
+            if (errorCode == S_OK && isSuccessful)
+            {
+              message = L"Printing " + std::wstring(title) +
+                              L" document to printer is succedded";
+            }
+            else
+            {
+              if (errorCode == S_OK)
+              {
+                message = L"Selected printer is not found, not available, offline or error state.";
+              }
+              else if (errorCode == E_INVALIDARG)
+              {
+                message = L"Invalid settings provided for the specified printer";
+              }
+              else if (errorCode == E_ABORT)
+              {
+                message = L"Printing " + std::wstring(title) +
+                                  L" document already in progress";
+              }
+              else if (errorCode == E_FAIL)
+              {
+                message = L"Printing " + std::wstring(title) +
+                                  L" document to printer is failed";
+              }
+            }
+
+            if (appWindow)
+            {
+              appWindow->AsyncMessageBox(message, L"Print to printer");
+            }
+            return S_OK;
+          })
+          .Get()));
+  return true;
+}
+```
+
+```c#
+async void PrintToPrinter()
+{
+  string printerName = GetPrinterName();
+  PrintSettings printInfo = GetSelectedPrinterPrintSettings(printerName);
```

Comment:
Why isn't printInfo a CoreWebView2PrintSettings? Why do you have your own PrintSettings class?

Reply:
Removed PrintSettings class